File: src/main/java/com/hirrao/NHCraft/utils/Mods.java
package com.hirrao.NHCraft.utils;

import cpw.mods.fml.common.Loader;

public enum Mods {

    ProgrammableHatches(Names.PROGRAMMABLE_HATCHES);

    public static class Names {

        public static final String PROGRAMMABLE_HATCHES = "programmablehatches";
    }

    public final String ID;
    private Boolean modLoaded;

    Mods(String ID) {
        this.ID = ID;
    }

    public boolean isModLoaded() {
        if (this.modLoaded == null) {
            this.modLoaded = Loader.isModLoaded(ID);
        }
        return this.modLoaded;
    }
}

File: src/main/java/com/hirrao/xnntech/common/xmod/enums/Mods.kt
package com.hirrao.xnntech.common.xmod.enums

import cpw.mods.fml.common.Loader

enum class Mods(val id: String) {
    ProgrammableHatches(Names.PROGRAMMABLE_HATCHES);

    var isModLoaded: Boolean? = null
        get() {
            if (field == null) {
                field = Loader.isModLoaded(id)
            }
            return field
        }
        private set

    object Names {
        const val PROGRAMMABLE_HATCHES: String = "programmablehatches"
    }
}
